src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -579,7 +579,8 @@ Copyright (c) .NET Foundation. All rights reserved.
       <ResolvedFileToPublish Remove ="%(ResolvedFileToRemove.Identity)"/>
 
       <ResolvedFileToPublish Include="%(NativeAppHostNETCore.Identity)">
-        <RelativePath>$(AssemblyName)%(Extension)</RelativePath>
+        <RelativePath>$(AssemblyName)</RelativePath>
+        <RelativePath Condition=" $(RuntimeIdentifier.StartsWith('win')) ">$(AssemblyName)%(Extension)</RelativePath>
```

Comment:
It is odd that %(Extension) is already the extension of the assembly name here.

I debugged it and it is because NativeAppHostNETCore already gets the name of the app [here](https://github.com/dotnet/sdk/blob/cd25d6bbfc2aac61cc790c511900427ac64fdaa8/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets#L271).

But that is only for 2.0+ apps. The code is quite convoluted. :(

I think the better targeted fix here is:

``` diff
     <ResolvedFileToPublish Include="%(NativeAppHostNETCore.Identity)">
-        <RelativePath>$(AssemblyName)%(Extension)</RelativePath>
+        <RelativePath>$(AssemblyName)$(_NativeExecutableExtension)</RelativePath>
```

That at least leaves the handling of executable extensions by platform in [one place](https://github.com/dotnet/sdk/blob/cd25d6bbfc2aac61cc790c511900427ac64fdaa8/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets#L64).